engine/apps/alerts/models/alert_group.py

```diff
@@ -401,15 +401,23 @@ def is_alert_a_resolve_signal(self, alert):
         raise NotImplementedError
 
     @property
-    def permalink(self):
+    def slack_permalink(self):
         if self.slack_message is not None:
             return self.slack_message.permalink
 
+    @property
+    def telegram_permalink(self) -> typing.Optional[str]:
+        main_telegram_message = self.telegram_messages.all()
+
+        if main_telegram_message:
+            return main_telegram_message[0].link
```

Comment:
This could potentially return a wrong telegram message if invoked directly (without any prefetching).
One of the ways to solve it would be to prefetch to a custom attribute in the Prefetch and check if this attribute is set in this method.

So the prefetch would be something like this `Prefetch("telegram_messages", ..., to_attr="prefetched_telegram_messages")`
And in the telegram_permalink property add a check for it: `if hasattr(self, "prefetched_telegram_messages"):`

But there's also a method that could be even prettier: https://stackoverflow.com/a/19651840. I have not tried this myself, but looks promising in this case.

Reply:
@vadimkerr thanks for pointing this out 😄

I believe [this commit](https://github.com/grafana/oncall/pull/551/commits/fd2d815d5823f61f3fc95dd303340d6494660c55) should address this concern.